python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -664,7 +815,11 @@ def dcrAssembleSubregion(self, dcrModels, subExposures, bbox, dcrBBox, warpRefLi
         residualGeneratorList = []
 
         for warpExpRef in warpRefList:
-            exposure = subExposures[warpExpRef.dataId["visit"]]
+            if isinstance(warpExpRef, DeferredDatasetHandle):
+                visit = warpExpRef.datasetRefOrType.dataId["visit"]
```

Comment:
datasetRefOrType is not guaranteed to have a dataId attribute, or it might be None. If you want an error message other than "str has no attribute dataId" or something like that, you should either try catch, or check dataId here. (in practice it will almost certainly do what you want, but the check will guard against some change somewhere else)

Reply:
That's good to know, and we should fix this since that error message would be would be hard for a user to debug. We use dataId in the same way all throughout `assembleCoaddTask` which this inherits from, so I think the solution would be to define a simple helper function there and use it in all of the assemble* tasks. Do you mind if I do that in a separate (very simple) ticket?

Reply:
no problem